lib/services/credentials/credential.dao.js

```diff
@@ -145,20 +145,37 @@ dao.removeCredential = function (id, type) {
   return db.del(buildIdKey(type, id));
 };
 
-dao.removeAllCredentials = function (id) {
+/*
+ * Remove all credentials
+ *
+ * @id {String}
+ * @deleteAll {Boolean} Optional - Whether to delete the key / set that contains all creds for users.
+ */
+dao.removeAllCredentials = function (id, deleteAll) {
   const dbTransaction = db.multi();
   const credentialTypes = Object.keys(config.models.credentials.properties);
-  const awaitAllPromises = [];
-  credentialTypes.forEach((type) => {
+  const awaitAllPromises = credentialTypes.map((type) => {
```

Comment:
If it's a single parameter, you can avoid the parentheses

```suggestion
  const awaitAllPromises = credentialTypes.map(type => {
```

Reply:
👍 for the `map`